test/aliases.js

```diff
@@ -0,0 +1,267 @@
+import test from 'ava'
+import { graphql } from 'graphql'
+import schema from '../test-api/schema-basic'
+import { partial } from 'lodash'
+import { errCheck } from './_util'
+
+const run = partial(graphql, schema)
+
+test('it should resolve aliases on different nesting levels', async t => {
+  const { data, errors } = await run(`
+    query {
+      aliasedUser: user(id: 1) {
+        aliasedFullName: fullName
+        aliasedPosts: posts {
+          aliasedId: id
+          aliasedAuthor: author {
+            aliasedFullName: fullName
+          }
+        }
+        aliasedFollowing: following(name: "matt") {
+          aliasedFullName: fullName
+        }
+      }
+    }
+  `)
+
+  errCheck(t, errors)
+
+  t.deepEqual({
+    aliasedFullName: 'andrew carlson',
+    aliasedPosts: [
+      {
+        aliasedId: 2,
+        aliasedAuthor: {
+          aliasedFullName: 'andrew carlson'
+        }
+      }
+    ],
+    aliasedFollowing: [
+      { aliasedFullName: 'matt elder' }
+    ],
+  }, data.aliasedUser)
+})
+
+test('it should allow an alias to the same relation (without args, same fields)', async t => {
+  const { data, errors } = await run(`
+    query {
+      user(id: 3) {
+        following1: following { fullName }
+        following2: following { fullName }
+      }
+    }
+  `)
+
+  errCheck(t, errors)
+  t.deepEqual({
+    following1: [
+      { fullName: 'andrew carlson' },
+      { fullName: 'matt elder' }
+    ],
+    following2: [
+      { fullName: 'andrew carlson' },
+      { fullName: 'matt elder' }
+    ],
+  }, data.user)
+})
+
+test.only('it should handle different args nested within aliases to the same relation', async t => {
```

Comment:
Did you use `test.only` on purpose here?

Reply:
Oops, good catch! Removed.